### DISCUSSION TITLE
How to split up the nextcloud app data location from lazy storage after inital setup?

### DISCUSSION THREAD

**Discussion by Wezea** [nextcloud/all-in-one#4729] [Questions] [answered]
Hi there,

I have all my data on an HDD and everything works just fine. However, for better performance reasons, I would like to separate e.g. all preview images (ideally the entire nextcould instance) to my local ssd. I found [this comment](https://github.com/nextcloud/previewgenerator/issues/391#issuecomment-1558291404) by @pulsejet  in an issue of the previewgenerator.

My question is:

Is it possible to alter this on an already initialized AIO setup, by just removing the container, set up the new mount points and the volumes like so and re-initialize?

```yml
    volumes:
      - /faststorage/nextcloud:/var/www/html
      - /raidarray/ncdata:/var/www/html/data
      - /faststorage/nc_appdata_oc213fxr1cuh:/var/www/html/data/nc_appdata_oc213fxr1cuh
```
Thanks a lot!

**Comment by szaimen** [comment]
> I would like to separate (ideally the entire nextcould instance) to my local ssd

You can do this: See https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-change-the-default-location-of-nextclouds-datadir and/or https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-store-the-filesinstallation-on-a-separate-drive

**Reply by Wezea** [reply]
Hi @szaimen thank you for your support!

I went through these suggestions already but they are not helpful for me. 

[This hint](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-change-the-default-location-of-nextclouds-datadir) is even somehow inconsistent. The first paragraph explicitly says don't do this after the first initialisation, but later it says _[...]If it was started already, you will need to stop the mastercontainer, remove it (no data will be lost)[...]_. 

[The second hint](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-store-the-filesinstallation-on-a-separate-drive) does also not tell how to split up runtime and app data storage from data storage. 

I am new to the docker world and might miss some fundamental details about how volumes work. 

I would appreciate some hints, how I could step by step, split up these locations on an already initialised instance.

Thank you again for the support and the hard work!

**Comment by szaimen** [comment, marked as answer]
Sorry that I was not clear enough.

> How to split up the nextcloud app data location from lazy storage after inital setup?

You cannot with "normal" AIO.

However you might make it work with https://github.com/nextcloud/all-in-one/tree/main/manual-install.

Further documentation for "normal" AIO: https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-change-the-default-location-of-nextclouds-datadir and https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-store-the-filesinstallation-on-a-separate-drive
